Documentation/design-docs/first-class-structs.md

```diff
@@ -0,0 +1,645 @@
+First Class Structs
+===================
+
+Objectives
+----------
+Primary Objectives
+* Avoid forcing structs to the stack if they are only assigned to/from, or passed to/returned
+ from a call or intrinsic
+ - Including SIMD types as well as other pointer-sized-or-less struct types
+ - Enable enregistration of structs that have no field accesses
+* Optimize these types as effectively as any other basic type
+ - Value numbering, especially for types that are used in intrinsics (e.g. SIMD)
+ - Register allocation
+
+Secondary Objectives
+* No “swizzling” or lying about struct types – they are always struct types
```

Comment:
Does this include removal of assumptions about how RuntimeType/Method/FieldHandle looks like and being able to represent them as normal structs within the JIT without code quality penalties?

Things like: https://github.com/dotnet/coreclr/blob/ef8af95b971a79fa0714fbeb39653a7581b4373f/src/jit/importer.cpp#L12548 - it would be nice for the underlying runtime to have control over the underlying representation of RuntimeTypeHandle.